src/Home/index.tsx

```diff
@@ -1,28 +1,73 @@
-import React, { useState } from 'react';
-import { FontAwesome } from '@expo/vector-icons';
+import React, { useEffect, useState } from 'react';
 import {
   AddIcon,
   Box,
   Circle,
   Divider,
   HStack,
   SearchIcon,
+  Spinner,
   useToast,
   VStack,
+  Text,
+  Button,
+  Stack,
+  Center,
 } from 'native-base';
 import { ScrollView } from 'react-native';
 import { Tile } from './Tile';
 import { ModalGameJoin } from './ModalGameJoin';
 import { API_URL } from '../constants';
 import { Game } from '../interface';
 import { useNavigation } from '../navigation/useNavigation';
+import { useIsFocused } from '@react-navigation/native';
+import { UserGame } from '../GameSelector/UserGame';
 
 export function Home({ navigation, player }: any) {
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const [showModal, setShowModal] = useState<boolean>(false);
+  const [userLastGames, setUserLastGame] = useState<Game[]>();
   const toast = useToast();
   const { navigateToGameState, navigateToGameSelector } =
     useNavigation(navigation);
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    getLastGames();
+  }, [isFocused]);
+
+  const getLastGames = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(
+        `${API_URL}/games?range=${encodeURI(
```

Comment:
It's ugly to have this kind of logic exposed.

Reply:
I would add a dataProvider to realize api calls are we have time